### PR TITLE
Setup: move ArtifactBuilder to Setup

### PR DESCRIPTION
**Pull request by klees**
Hi everybody!

As discussed in PR #1862 this moves the ArtifactBuilder from @chfsx to the Setup. This also contains some improvements and corrections in the current Setup-Code I discovered along the way.

This neither contains breaking changes to existing interfaces nor any other stuff that needs to be approved by the JF. I simply show this in the open to give @chfsx and other interested developers a chance to notice this change and voice their opinions.

If I get no feedback here, I will merge this on 2019-07-17.

Best regards!

### CODE REVIEW

**Comment by klees** [comment]
The build currently fails. This seems to be caused by some, probably dead, code, that uses classes from PEAR. I'm trying to solve this with the responsible maintainer.

File: setup/classes/class.ilSetupLanguage.php [hunk 1]
```diff
@@ -41,7 +43,7 @@
  *  The format functions do not belong in class.Language. Those are also applicable elsewhere.
  *  Therefore, they would be better placed in class.Format
  */
-class ilLanguage
+class ilSetupLanguage extends ilLanguage
```
Comment (chfsx) [inline setup/classes/class.ilSetupLanguage.php:46]: @klees this is not related to the Artifact-thingy, isn't it? Absolutely fine for me, but there seem to be different declarations e.g. for `txt(...)`.

File: setup/cli.php [hunk 2]
```diff
@@ -43,6 +57,12 @@ function build_container_for_setup() {
 		);
 	};
 
+	$c["agent.global_screen"] = function($c) {
```
Comment (chfsx) [inline setup/cli.php:60]: @klees on the long term, wouldn't be great to have some kind of `Collector` which parses the code-base to collect all Agents?

**Review by chfsx — APPROVED** [review]
@klees some questions and remarks in the code, but LGTM in general! Thanks!

File: setup/classes/class.ilSetupLanguage.php (hunk 1 above)
Comment (klees) [reply setup/classes/class.ilSetupLanguage.php:46]: PHP yelled at me, because both classes have been loaded the same time when I used that Artifact-thingy. So its kind of related, but also not exactly =)

I expect this to be an intermediate solution, and you are of course absolutely right regarding the declaration. This is some of the technical depth in the setup that needs to be fixed.

File: setup/cli.php (hunk 2 above)
Comment (klees) [reply setup/cli.php:60]: Yes., absolutely.

Luckily you already contributed that `ImplementationOfInterfaceFinder`. I'm just a little reluctant to do this just now, I want to get some other stuff right before I pull in complete ILIAS =)

**Comment by klees** [comment]
The dead code is still included, I instead just ignore it for now:

https://github.com/ILIAS-eLearning/ILIAS/pull/1990/commits/48ad4ba12210b8cf9ee0041d9cac9eb9f323fa02
